eng/versioning.targets

```diff
@@ -24,19 +24,23 @@
 
   <!-- Adds SupportedOSPlatform or UnsupportedOSPlatform attributes -->
   <ItemGroup Condition="'$(IsTestProject)' != 'true'">
-    <_supportedOSPlatforms Include="$(SupportedOSPlatforms)" />
-    <_unsupportedOSPlatforms Include="$(UnsupportedOSPlatforms)" />
+    <_supportedOSPlatforms Include="$(AssemblySupportedOSPlatformAttributes)" />
+    <_unsupportedOSPlatforms Include="$(AssemblyUnsupportedOSPlatformAttributes)" Condition="'$(_assemblySupportedOSPlatformAttributesOverride)' != 'true'" />
   </ItemGroup>
 
   <Target Name="AddAssemblyOSPlatformAttributes" BeforeTargets="GenerateAssemblyInfo" AfterTargets="PrepareForBuild">
     <ItemGroup>
+      <!-- When there are Supported OSPlatforms, transform each of those into an attribute -->
       <AssemblyAttribute Include="System.Runtime.Versioning.SupportedOSPlatform" Condition="'@(_supportedOSPlatforms)' != ''">
         <_Parameter1>%(_supportedOSPlatforms.Identity)</_Parameter1>
       </AssemblyAttribute>
+      <!-- When there are not any Supported OSPlatforms, but there are Unsupported OSPlatforms, transform each of those into an attribute -->
       <AssemblyAttribute Include="System.Runtime.Versioning.UnsupportedOSPlatform" Condition="'@(_supportedOSPlatforms)' == '' and '@(_unsupportedOSPlatforms)' != ''">
         <_Parameter1>%(_unsupportedOSPlatforms.Identity)</_Parameter1>
       </AssemblyAttribute>
     </ItemGroup>
+    <Warning Condition="'@(_supportedOSPlatforms)' != '' and '@(_unsupportedOSPlatforms)' != ''"
```

Comment:
I would just Error. We set warnAsError for the global build, so it is the same, we usually try to error for this kind of stuff as warnings could slip sometimes.